packages/devtools_app/lib/src/inspector/flutter/data_models.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import '../enum_deserializer.dart';
+
+const Type boxConstraintsType = BoxConstraints;
+
+Constraints deserializeConstraints(Map<String, Object> json) {
+  // TODO(albertusangga): Support SliverConstraint
+  if (json == null || json['type'] != boxConstraintsType.toString())
+    return null;
+  return BoxConstraints(
+    minWidth: json['minWidth'],
+    maxWidth: json['hasBoundedWidth'] ? json['maxWidth'] : double.infinity,
+    minHeight: json['minHeight'],
+    maxHeight: json['hasBoundedHeight'] ? json['maxHeight'] : double.infinity,
+  );
+}
+
+/// TODO(albertusangga): Move this to [RemoteDiagnosticsNode] once dart:html app is removed
+@immutable
+class RenderFlexProperties {
```

Comment:
one possible idea to improve this is to reuse the `RenderFlex` class from `Flutter` since we can reuse the fields and not redeclare them, but to be honest I'm still pretty unsure about this, or do you think that the current approach is better?

Reply:
It is fine to keep this separate from RenderFlex. We aren't trying to attach this to a real RenderObject tree :)